dali/base/sysinfologger.cpp

```diff
@@ -23,42 +23,49 @@
 #define SYS_INFO_VERSION "1.0"
 
 #define SYS_INFO_ROOT  "/SysLogs"
-#define ATTR_NEXTID    "@nextId"
+#define ATTR_NEXTSEQN  "@nextSeqNum"
 #define ATTR_VERSION   "@version"
 #define MSG_NODE       "msg"
-#define ATTR_ID        "@id"
+#define ATTR_SEQNUM    "@seqNum"
 #define ATTR_TIMESTAMP "@ts"
 #define ATTR_AUDIENCE  "@audience"
 #define ATTR_CLASS     "@class"
 #define ATTR_CODE      "@code"
 #define ATTR_HIDDEN    "@hidden"
 #define ATTR_SOURCE    "@source"
 
+static constexpr unsigned __int64 nonSysInfoLogMsgMask = 0x80000000;
+
 static void extractDate(timestamp_type ts, unsigned & year, unsigned & month, unsigned & day)
 {
     CDateTime timeStamp;
     timeStamp.setTimeStamp(ts);
     timeStamp.getDate(year, month, day);
 }
 
-static unsigned __int64 makeMessageId(unsigned year, unsigned month, unsigned day, unsigned id)
+unsigned __int64 makeMessageId(unsigned year, unsigned month, unsigned day, unsigned seqN, bool nonSysInfoLogMsg)
 {
-    return id<<21 | year<<9 | month<<5 | day;
+    // bits 0-4 = day, bits 5-8 = month, bits 9-22 = year
+    // bit 23-30 unused
+    // bit 31 = (flag) unset means SysInfoLogger managed message, set means message managed elsewhere
+    // bits 32-63 = sequence number
+    return seqN<<32 | (nonSysInfoLogMsg?nonSysInfoLogMsgMask:0) | year<<9 | month<<5 | day;
```

Comment:
Does this work?  seqN is an unsigned seqn<<32 will be 0 (or undefined).

Reply:
Are you concerned about seqn being > 32 bits? @ghalliday

Reply:
It works.  Should I make all the vars 64bit to make it clearer?

Reply:
I am not sure what you mean by it works.  It didn't even compile on my machine in debug when I pasted it into the code generator:

: error: left shift count >= width of type [-Werror=shift-count-overflow]

It looks correct with the extra cast.